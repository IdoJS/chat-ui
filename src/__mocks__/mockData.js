const MOCK_USER_NAME = 'mock_user_name';
const MOCK_AVATAR_CLASS = 'mock_avatar_class';
const MOCK_TEXT = 'mock_text';
const MOCK_SOCKET_REQ = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: MOCK_USER_NAME, userId: '0'};
const MOCK_SELF_MESSAGE = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: MOCK_USER_NAME, timestamp: 0, userId: '0'};
const MOCK_OTHER_MESSAGE = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: 'other_user', timestamp: 0, userId: '1'};

const MOCK_CHAT_DATA_INIT = {
  loading: false,
  timestamp: Date.now(),
  chatArr: [],
  typingArr: [],
  timeout: false
};
export {
  MOCK_USER_NAME,
  MOCK_AVATAR_CLASS,
  MOCK_TEXT,
  MOCK_SOCKET_REQ,
  MOCK_CHAT_DATA_INIT,
  MOCK_SELF_MESSAGE,
  MOCK_OTHER_MESSAGE
}