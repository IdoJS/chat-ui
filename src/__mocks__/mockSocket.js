// modify the code https://medium.freecodecamp.org/testing-socket-io-client-app-using-jest-and-react-testing-library-9cae93c070a3

let EVENTS = {};
function emit(event, ...args) {
  EVENTS[event].forEach(func => func(...args));
}
const socket = () => {
  return {
    on(event, func) {
      if (EVENTS[event]) {
        return EVENTS[event].push(func);
      }
      EVENTS[event] = [func];
    },
    emit
  }
};
export const mockIo = {
  connect() {
    return socket;
  }
};

export default mockIo;