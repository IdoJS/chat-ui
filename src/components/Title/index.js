import React from 'react';
import PropTypes from 'prop-types';

const Title = (props) => {
  return <h1 className="secondary_title">Hello {props.userName}</h1>;
};

Title.propTypes = {
  userName : PropTypes.string.isRequired
};

Title.defaultProps = {
  userName : ''
};

export default Title