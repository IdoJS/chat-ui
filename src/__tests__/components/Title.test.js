import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import {MOCK_USER_NAME} from '../../__mocks__/mockData';

import Title from '../../../src/components/Title';

describe('Title component', () => {
  it('Should render the Title view', () => {

    const wrapper = shallow(<Title userName={MOCK_USER_NAME}/>);
    const title = wrapper.find('h1');

    expect(title).toHaveLength(1);

  });

  it('SnapShot rendering ChatRoom', () => {
    const wrapper = shallow(<Title userName={MOCK_USER_NAME}/>);
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

});