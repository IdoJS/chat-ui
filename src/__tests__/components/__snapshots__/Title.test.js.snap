// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component SnapShot rendering ChatRoom 1`] = `
<h1
  className="secondary_title"
>
  Hello 
  mockUserName
</h1>
`;
